### PR TITLE
Solution 07 Memory Management module

### PR DESCRIPTION
**Pull request by pahomov-and**
Signed-off-by: Andrey Pahomov <pahomov.and@gmail.com>

### CODE REVIEW

File: 07-MemoryManagement/module/module_07.c [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+#include <linux/uaccess.h>
+#include <linux/device.h>
+
+#include <linux/tick.h>
+#include <linux/mm.h>
+#include <linux/slab.h>
+
+#define MAX_POW 21
+#define INTERVAL_1NS 1000000000ULL
+#define POW2(x) (2 << (unsigned long)(x))
```
Comment (yekovalyov) [inline 07-MemoryManagement/module/module_07.c]: 2UL << x  looks better in my opinion.

File: 07-MemoryManagement/module/log.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# uname -a
+Linux qemuLinux 4.13.0 #1 SMP Tue Oct 22 22:51:53 EEST 2019 i686 GNU/Linux
+# cat /sys/module_07/mem_test 
+size/dt     kmalloc     kfree       vmalloc     vfree       kzmalloc    kzfree      getFreePage freePages   
+2           31151       27605       52754       21949       613110      16350       805902      46313
```
Comment (yekovalyov) [inline 07-MemoryManagement/module/log.txt]: Is the size of kmalloc and getFreePages the same?

File: 07-MemoryManagement/module/log.txt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+4096        700905      1599        105088      4641        695782      12734       0           0           
+8192        2759        955         9196        3769        31309       164781      0           0           
+16384       692158      7242        9442        4981        682997      48001       0           0           
+32768       5397        2388        14628       5536        74425       82949       0           0           
+65536       3856        1890        16202       7197        80794       180556      0           0           
+131072      1016598     5067        22671       16611       155609      423206      0           0           
+262144      5315        2959        44751       27350       1027180     713962      0           0           
+524288      5230        3466        69317       61921       424667      1376990     0           0           
+1048576     9517        5642        2037189     126985      1422670     2846720     0           0           
+2097152     11108       8519        261892      221231      2409655     7006185     0           0
```
Comment (yekovalyov) [inline 07-MemoryManagement/module/log.txt]: Why time for get_free_pages is 0?

**Comment by yekovalyov** [comment]
Please add option -Wall for gcc and fix warnings.

**Comment by yekovalyov** [comment]
Please check code using checkpatch.pl
kernel/linux/scripts/checkpatch.pl --no-tree -f main.c

**Comment by yekovalyov** [comment]
Please also provide the results in text format.

File: 07-MemoryManagement/module/x86_64_out.txt [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+8           158         193         1025        427         190         193         157         98          
+8           293         122         488         319         90          98          78          56          
+8           295         162         960         900         197         167         160         102         
+8           78          69          684         324         92          86          79          46          
+16          1237        772         48181       3360        10438       9244        158         97          
+16          1290        812         76243       3395        13413       9227        158         100         
+16          1416        939         61797       1965        6689        4573        79          47          
+16          1545        760         93206       3690        12790       9187        157         100         
+16          755         404         58259       1643        5481        4679        79          55          
+32          0           0           0           0           0           0           157         115
```
Comment (yekovalyov) [inline 07-MemoryManagement/module/x86_64_out.txt]: Is the size of memory allocated by get_free_pages greater than size of vmalloc?

Comment (pahomov-and) [reply 07-MemoryManagement/module/x86_64_out.txt]: Fixed bug. I set the size of the allocated memory incorrectly.

File: 07-MemoryManagement/module/log.txt (hunk 3 above)
Comment (yekovalyov) [reply 07-MemoryManagement/module/log.txt]: Thank You for fixing this issue.

**Review by yekovalyov — APPROVED** [review]
Nice implementation. Finally the bug is fixed.

**Pull request merged**